stories/patterns/ux-patterns/form/docs/examples/default-log-in-form.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import { FormProvider, useForm, useFormContext, RegisterOptions } from 'react-hook-form';
+import { Flex } from '@semcore/flex-box';
+import Tooltip from '@semcore/tooltip';
+import Input from '@semcore/input';
+import Button from '@semcore/button';
+import { Text } from '@semcore/typography';
+
+type FormValues = {
+  email: string;
+  password: string;
+};
+
+const defaultValues = { email: '', password: '' };
+
+type FormControlProps = {
+  name: keyof FormValues;
+  type: string;
+  options: RegisterOptions;
+};
+
+const FormControl = ({ name, type, options }: FormControlProps) => {
+  const {
+    register,
+    trigger,
+    getFieldState,
+    resetField,
+    getValues,
+    formState: { isSubmitted, errors },
+  } = useFormContext();
+  const { isTouched } = getFieldState(name);
+  const [active, setActive] = React.useState<boolean>(false);
+  const error = errors[name];
+
+  const hasError = () => {
+    if (error?.type === 'required' && !isSubmitted) {
+      return false;
+    }
+
+    return Boolean(error);
+  };
+
+  const invalid = (): boolean => {
+    return isTouched && hasError();
+  };
+
+  const showErrorTooltip = (): boolean => {
+    return invalid() && active;
+  };
+
+  const resetIfChangedToValid = (isValid: boolean) => {
+    if (isValid) {
+      resetField(name, { keepTouched: false, defaultValue: getValues(name) });
+    }
+  };
+
+  const { onChange, ...restField } = register(name, {
+    ...options,
+    onBlur: () => setActive(false),
+  });
+
+  const field = {
+    onChange: (_v: string, e: React.SyntheticEvent) => {
+      // important: keep call order, otherwise validation breaks
+      onChange(e);
+      hasError() && trigger().then(resetIfChangedToValid);
+    },
+    ...restField,
+  };
+
+  return (
+    <Tooltip placement='top' interaction={'none'} animationsDisabled>
+      <Tooltip.Popper visible={showErrorTooltip()} id={`form-${name}-error`} theme='warning'>
+        {error?.message as any}
+      </Tooltip.Popper>
+      <Tooltip.Trigger
+        tag={Input}
+        w='100%'
+        mb={2}
+        size='l'
+        state={invalid() ? 'invalid' : 'normal'}
+        controlsLength={1}
+      >
+        {({ getTriggerProps }) => {
+          return (
+            <Input.Value
+              {...getTriggerProps({
+                id: name,
+                type: type,
+              })}
+              {...field}
+              onFocus={() => setActive(true)}
+              autoComplete={type}
+              aria-invalid={invalid()}
+              aria-errormessage={invalid() ? `form-${name}-error` : undefined}
+            />
+          );
+        }}
```

Comment:
``` {({ getTriggerProps }) => {``` should be in `<Tooltip>` and I think, you don't need `<Tooltip.Trigger` at all

Reply:
I just kept original structure of code i copy-pasted from previous version. This change is not related to the issue mentioned in the JIRA. Maybe we can keep as it is and address this issue in the future?

Reply:
Fixed